### PR TITLE
[WEB-784] Algolia API-focused docsearch updates

### PR DESCRIPTION
**Pull request by bgdeutsch**
### What does this PR do?
Adds a replica docsearch index that is used on `api` pages to bump API-specific search results to the top.
Adjusts some styles and algolia autocomplete behavior to highlight API-specific results.

### Motivation
https://datadoghq.atlassian.net/browse/WEB-784

### Preview
https://docs-staging.datadoghq.com/brian.deutsch/api-algolia-search/api/

- `API` / `API Latest` search results should be at the top of the algolia widget on `/api` pages.
- there should be a link `Click here to search the full docs` on `/api` pages.   this links to the `/search` page with query parameter appended.
- left-side column of algolia widget should have the API endpoints on `/api` pages.
- Home & non-API page algolia search should be unaffected by these updates.

synthetics: https://dd-corpsite.datadoghq.com/synthetics/details/ycq-nxu-nkx

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: local/etc/algolia/docs_datadoghq_preview.json [hunk 1]
```diff
@@ -136,9 +136,9 @@
     "selectors": {
         "api_latest": {
             "lvl0": {
-                "selector": "",
+                "selector": ".h-100 .sidenav-nav > ul >li.open > a",
                 "global": true,
-                "default_value": "API Latest"
+                "default_value": "API"
```
Comment (davidejones) [inline local/etc/algolia/docs_datadoghq_preview.json:141]: I think you will need to add the same changes to `docs_datadoghq.json`

Comment (bgdeutsch) [reply local/etc/algolia/docs_datadoghq_preview.json:141]: thanks @davidejones!

**Review by kayayarai — APPROVED** [review]

**Review by nsollecito — APPROVED** [review]
👍

**Comment by apigirl** [comment]
David is 🟢 to merge in his morning.

**Pull request merged**